src/android/ImagePicker.java

```diff
@@ -0,0 +1,204 @@
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import android.Manifest;
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.ClipData;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.provider.MediaStore;
+import android.provider.OpenableColumns;
+import android.util.Log;
+import android.widget.Toast;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+
+import androidx.core.content.ContextCompat;
+
+public class ImagePicker extends CordovaPlugin {
+    private static final String ACTION_GET_PICTURES = "getPictures";
+    private static final String ACTION_HAS_READ_PERMISSION = "hasReadPermission";
+    private static final String ACTION_REQUEST_READ_PERMISSION = "requestReadPermission";
+
+    private static final int PERMISSION_REQUEST_CODE = 100;
+    private static final int SELECT_PICTURE = 200;
+
+    private CallbackContext callbackContext;
+    private int maxImageCount;
+
+    public boolean execute(String action, final JSONArray args, final CallbackContext callbackContext) throws JSONException {
+        this.callbackContext = callbackContext;
+        if (ACTION_HAS_READ_PERMISSION.equals(action)) {
+            callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK, hasReadPermission()));
+            return true;
+
+        } else if (ACTION_REQUEST_READ_PERMISSION.equals(action)) {
+            requestReadPermission();
+            return true;
+
+        } else if (ACTION_GET_PICTURES.equals(action)) {
+
+            final JSONObject params = args.getJSONObject(0);
+            this.maxImageCount = params.has("maximumImagesCount") ? params.getInt("maximumImagesCount") : 20;
+
+            Intent imagePickerIntent = new Intent(Intent.ACTION_OPEN_DOCUMENT, MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+            imagePickerIntent.setType("image/*");
+            imagePickerIntent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
+
+            cordova.startActivityForResult(this, imagePickerIntent, SELECT_PICTURE);
+            this.showLimitExceededMessage();
+            return true;
+        }
+        return false;
+    }
+
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (resultCode == Activity.RESULT_OK && data != null) {
+            ArrayList<String> fileURIs = new ArrayList<String>();
+            if (requestCode == SELECT_PICTURE) {
+                if (data.getData() != null) {
+                    Uri uri = data.getData();
+                    String path = this.copyFileToInternalStorage(uri, "");
+                    if (path.equals("-1")) {
+                        callbackContext.error("For the moment cross sharing of media between work profile and personal profile is not supported.");
```

Comment:
On peut mettre le texte dans une variable.
```suggestion
                        callbackContext.error("Cannot access file. (-1)");
```
?
On améliore les error codes au fur et à mesure.